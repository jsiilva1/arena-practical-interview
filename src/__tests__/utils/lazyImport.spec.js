import React from 'react';

import { shallow } from 'enzyme';

import LazyImport from '../../shared/utils/lazyImport';

describe('Lazy Import Helper Util', () => {
  it('should return a wrapper of loading component of the requested component chunk', () => {
    const ReposPage = LazyImport({
      loader: () => import('../../containers/PullRequests' /* webpackChunkName: 'app-chunk-repos-page' */),
    });

    const wrapper = shallow(<ReposPage />);

    expect(wrapper).toMatchSnapshot();
  });
});