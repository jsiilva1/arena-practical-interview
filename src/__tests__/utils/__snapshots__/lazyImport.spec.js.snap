// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy Import Helper Util should mount the requested component chunk 1`] = `
<LoadingComponent
  error={null}
  isLoading={true}
  pastDelay={false}
  retry={[Function]}
  timedOut={false}
/>
`;

exports[`Lazy Import Helper Util should return a wrapper of loading component of the requested component chunk 1`] = `
<LoadingComponent
  error={null}
  isLoading={true}
  pastDelay={false}
  retry={[Function]}
  timedOut={false}
/>
`;
